src/rgw/rgw_aio.cc

```diff
@@ -93,6 +124,31 @@ Aio::OpFunc aio_abstract(Op&& op, boost::asio::io_context& context,
     };
 }
 
+
+template <typename Op>
+Aio::OpFunc cache_aio_abstract(Op&& op, off_t obj_ofs, off_t read_ofs, off_t read_len, std::string& location) {
+  return [op = std::move(op), obj_ofs, read_ofs, read_len, location] (Aio* aio, AioResult& r) mutable{
+    auto& ref = r.obj.get_ref();
+    auto cs = new(&r.user_data) cache_state(aio, r);
+    cs->c = new D3nL1CacheRequest();
+    if (g_conf()->rgw_d3n_l1_libaio_read) {
+      cs->c->prepare_libaio_op(ref.obj.oid, &r.data, read_len, obj_ofs, read_ofs, location, cache_aio_cb, aio, &r);
+      int ret = cs->submit_libaio_op(cs->c);
+      if(ret < 0) {
+        r.result = -1;
+        cs->aio->put(r);
+      }
+    } else {
+      int ret = cs->c->execute_io_op(ref.obj.oid, &r.data, read_len, obj_ofs, read_ofs, location, cache_aio_cb, aio, &r);
```

Comment:
will change the `rgw_d3n_l1_libaio_read` default val to `true`
(In quick benchmarks, there was no performance difference)

Reply:
is there any value in keeping the synchronous option? we have built-in concurrency, both from `Read::iterate()`'s use of `AioThrottle`, and from the beast frontend's use of coroutines to suspend instead of block a thread. the goal of the beast+async project is to remove all blocking from the op path so we can run with a small number of threads

Reply:
bump, please remove the non-aio version (and `rgw_d3n_l1_libaio_read` option)

Reply:
> bump, please remove the non-aio version (and rgw_d3n_l1_libaio_read option)

sure, removing...